src/rules/prefer-in-document.js

```diff
@@ -79,6 +85,14 @@ export const create = (context) => {
       }
     }
 
+    // toBe() or toEqual() are only invalid with null
+    if (
+      (matcherNode.name === "toBe" || matcherNode.name === "toEqual") &&
+      !usesToBeOrToEqualWithNull(matcherNode, matcherArguments)
+    ) {
```

Comment:
I merged the 2 if in one condition. I believe this is _less_ readable, not more, but well let's see if eslint is happier with this.